api/src/main/java/net/kyori/adventure/text/JoinConfiguration.java

```diff
@@ -0,0 +1,222 @@
+ *   <dt>an operator (non-optional)</dt>
+ *   <dd>a unary operator to change each component that is being joined, defaults to the identity operator</dd>
+ * </dl>
+ *
+ * <p>Note that the last separator only acts as an override for the normal separator.
+ * This means that if you do not specify a last separator, the normal separator will be placed between the last two components.
+ * To omit the final separator, but still include normal separators, use {@link Component#empty()} as the last separator.</p>
+ *
+ * @see Component#join(JoinConfiguration, Iterable)
+ * @see Component#join(JoinConfiguration, ComponentLike...)
+ * @since 4.8.0
+ */
+public interface JoinConfiguration extends Buildable<JoinConfiguration, JoinConfiguration.Builder>, Examinable {
+  /**
+   * Creates a new builder.
+   *
+   * @return a new builder
+   * @since 4.8.0
+   */
+  static @NonNull Builder builder() {
+    return new JoinConfigurationImpl.BuilderImpl();
+  }
+
+  /**
+   * Gets a join configuration with no separators, prefix or suffix.
+   *
+   * @return the join configuration
+   * @since 4.8.0
+   */
+  static @NonNull JoinConfiguration noSeparators() {
+    return JoinConfigurationImpl.NULL;
+  }
+
+  /**
+   * Creates a join configuration with a separator and no prefix or suffix.
+   *
+   * @param separator the separator
+   * @return the join configuration
+   * @since 4.8.0
+   */
+  static @NonNull JoinConfiguration separator(final @Nullable ComponentLike separator) {
+    if(separator == null) return JoinConfigurationImpl.NULL;
+    return builder().separator(separator).build();
+  }
+
+  /**
+   * Creates a join configuration with a separator and last separator but no prefix or suffix.
+   *
+   * @param separator the separator
+   * @param lastSeparator the last separator
+   * @return the join configuration
+   * @since 4.8.0
+   */
+  static @NonNull JoinConfiguration separators(final @Nullable ComponentLike separator, final @Nullable ComponentLike lastSeparator) {
+    if(separator == null && lastSeparator == null) return JoinConfigurationImpl.NULL;
+    return builder().separator(separator).lastSeparator(lastSeparator).build();
+  }
+
+  /**
+   * Gets the prefix of this join configuration.
+   *
+   * @return the prefix
+   * @since 4.8.0
+   */
+  @Nullable ComponentLike prefix();
+
+  /**
+   * Gets the suffix of this join configuration.
+   *
+   * @return the suffix
+   * @since 4.8.0
+   */
+  @Nullable ComponentLike suffix();
+
+  /**
+   * Gets the separator of this join configuration.
+   *
+   * @return the separator
+   * @since 4.8.0
+   */
+  @Nullable ComponentLike separator();
+
+  /**
+   * Gets the last separator of this join configuration.
+   *
+   * @return the last separator
+   * @since 4.8.0
+   */
+  @Nullable ComponentLike lastSeparator();
+
+  /**
+   * Gets the operator of this join configuration.
+   *
+   * @return the operator
+   * @since 4.8.0
+   */
+  @NotNull UnaryOperator<ComponentLike> operator();
```

Comment:
Should this be `UnaryOperator<? super ComponentLike>`?

Reply:
I'm not sure - what would adding this do?